aten/src/ATen/native/TensorShape.cpp

```diff
@@ -49,7 +49,73 @@ Tensor & cat_out(Tensor & result, TensorList tensors, int64_t dim) {
   return at::_th_cat_out(result, tensors, dim);
 }
 
+static bool sizes_match_except(IntList s1, IntList s2, int64_t dim_except) {
+  if (s1.size() != s2.size()) {
+    return false;
+  }
+  for (int64_t i = 0; i < s1.size(); ++i) {
+    if (i != dim_except && s1[i] != s2[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+static Tensor cat_sparse(TensorList tensors, int64_t dim) {
+  std::vector<Tensor> indices;
+  std::vector<Tensor> values;
+  int64_t sparse_dim = tensors[0].sparse_dim();
+  int64_t dense_dim = tensors[0].dense_dim();
+  AT_CHECK(dim < sparse_dim,
```

Comment:
could you add a comment about why this isn't supported?  I'm cool not doing it yet, but it doesn't seem impossible either.

Reply:
It is indeed not impossible -- I can add a comment (and make it work in a future PR)